api/src/test/java/org/openmrs/module/dtd/ConfigXmlBuilder.java

```diff
@@ -237,10 +242,15 @@ private void initDocument() throws ParserConfigurationException {
 		documentBuilder.newDocument();
 	}
 	
-	private void setDocType() {
+	private void setDocType() throws FileNotFoundException, URISyntaxException {
 		DOMImplementation domImpl = configXml.getImplementation();
-		DocumentType doctype = domImpl.createDocumentType(MODULE_NAME, PUBLIC_IDENTIFIER,
-				"https://resources.openmrs.org/doctype/config-" + dtdVersion + ".dtd");
+
+		URL dtdResource = ConfigXmlBuilder.class.getResource("/org/openmrs/module/dtd/config-" + dtdVersion + ".dtd");
```

Comment:
Is this a change specifically required by the addition of init parameters or just a general improvement? I personally prefer having a separate ticket for general improvements such that we limit the scope of changes to only what is required by the ticket.

Reply:
It isn't specific to init parameters, but I couldn't add dtd tests to this PR without this change. The reason is that the tests are currently using the tests on the server which I don't think usually get added to the server until after merging a PR right?

If you want them in separate PRs what I could do is create a PR that updates that DTD and tests and merge that first. Then in a separate PR we could merge the the servlet related changes. Would you prefer doing it that way?

Reply:
No problem. Let us just leave it as is. 😊